spec/models/initiative_spec.rb

```diff
@@ -26,4 +26,19 @@
     expect{ delete :destroy, params: {id: initiative.id}}.to change(Initiative,:count).by(-1)
     expect(response).to redirect_to root_path
   end
+end
+
+describe "#index" do
+  it "orders by initiative status" do
+    @initiative = Initiative.new(title: 'initiative1', target_date: "01/01/2020", completion: true)
+    @initiative.save
+    
+    @initiative1 = Initiative.new(title: 'initiative2', target_date: "01/01/2020")
+    @initiative1.save
+
+    initt = Initiative.order('completion DESC').all.to_a
+    init = [@initiative1, @initiative]
```

Comment:
This test never actually calls the `index` method. By evaluating `initt` and `init` against each other, you're actually only testing that Active Record's `order` method does what it's supposed to (which it does, it's been well tested in its own library :) ). Controller tests typically model http requests and responses, and assert that the response from the request matches an expected structure. So you could assert that a `response` object would equal a structure like `init`, but you'd need to call `index` to create that `response`. Check out https://relishapp.com/rspec/rspec-rails/docs/controller-specs for some examples. 

Also, consider the way you're naming variables. `init` doesn't communicate the value of the variable very well, but `initiatives` or `expected` would.

Reply:
The test has been altered, but not quite sure if it's entirely correct